fritzconnection/lib/fritzwlan.py

```diff
@@ -24,17 +24,51 @@
     SEGNO_INSTALLED = True
 
 # important: don't set an extension number here:
-SERVICE = 'WLANConfiguration'
+SERVICE = "WLANConfiguration"
 DEFAULT_PASSWORD_LENGTH = 12
-WPA_SECURITY = 'WPA'
-NO_PASS = 'nopass'
+
+
+class Security(Enum):
+    """Enum representing different security types for WLAN networks.
+
+    The Security enum class defines constants for various
+    security types used in WLAN networks.
+    Each constant represents a specific security configuration.
+
+    Constants:
+        WPA_TKIP_SECURITY: Security type using WPA with TKIP encryption.
+        WPA2_CCMP_SECURITY: Security type using WPA2 with CCMP/AES encryption.
+        WPA_WPA2_SECURITY: Security type using both WPA and WPA2 (mixed mode).
+        WPA_SECURITY: Deprecated constant for WPA security.
+        NO_PASS: Security type for open networks with no password (no passphrase).
+        NONE_SECURITY (NoneType): Represents no specific security.
+
+    Note:
+        - The WPA_SECURITY constant is duplicated and deprecated.
+          It has the same value as WPA_TKIP_SECURITY.
+        - The NO_PASS constant is used for open networks with no password required.
+        - The NONE_SECURITY constant represents the absence of a specific security configuration.
+
+    Example:
+        >>> network_security = Security.WPA2_CCMP_SECURITY
+        >>> print(network_security)
+        Security.WPA2_CCMP_SECURITY
+
+    """
+
+    WPA_TKIP_SECURITY = "WPA"
+    WPA2_CCMP_SECURITY = "11i"
+    WPA_WPA2_SECURITY = "WPAand11i"
+    WPA_SECURITY = "WPA"
+    NO_PASS_SECURITY = "nopass"
+    NONE_SECURITY = None
+
```

Comment:
Depending on the further use of this enumeration it is unclear to me whether thing are getting more complicated or not. Also this large diff (all three commits with partly lots of changes in a single pull-request) makes it hard to understand what's happening.